### PR TITLE
Improve static export documentation

### PR DESCRIPTION
**Pull request by leerob**
There has been some helpful [discussion on Twitter](https://twitter.com/dan_abramov/status/1636778278882099216) around this page of documentation. I refactored the introduction to make it more clear that you also get to use SPA features when doing a static export, and underscoring the value of being able to deploy/host virtually anywhere with this approach.

### CODE REVIEW

File: docs/advanced-features/static-html-export.md [hunk 1]
```diff
@@ -11,9 +11,11 @@ description: Export your Next.js app to static HTML, and run it standalone witho
   </ul>
 </details>
 
-`next export` allows you to export your Next.js application to static HTML, which can render without the need of a Node.js server. It is recommended to only use `next export` if you don't need any of the [unsupported features](#unsupported-features) requiring a server.
+Next.js can be used to generatic static applications, including using React in the browser without the need for a server.
```
Comment (styfle) [inline docs/advanced-features/static-html-export.md:14]: I don't think "without the need for a server" is correct.

What would serve the html besides a server?

Comment (styfle) [inline docs/advanced-features/static-html-export.md:14]:
```suggestion
Next.js can be used to generate static applications, including using React in the browser without the need for a Node.js server.
```

Comment (leerob) [reply docs/advanced-features/static-html-export.md]: That's good feedback, agreed on saying Node.js specifically. Thank you for the suggestion.

**Review by styfle — APPROVED** [review]

File: docs/advanced-features/static-html-export.md [hunk 2]
```diff
@@ -11,9 +11,11 @@ description: Export your Next.js app to static HTML, and run it standalone witho
   </ul>
 </details>
 
-`next export` allows you to export your Next.js application to static HTML, which can render without the need of a Node.js server. It is recommended to only use `next export` if you don't need any of the [unsupported features](#unsupported-features) requiring a server.
+Next.js can be used to generate static applications, including using React in the browser without the need for a Node.js server.
 
-If you're looking to build a hybrid site where only _some_ pages are prerendered to static HTML, Next.js already does that automatically. Learn more about [Automatic Static Optimization](/docs/advanced-features/automatic-static-optimization.md) and [Incremental Static Regeneration](/docs/basic-features/data-fetching/incremental-static-regeneration.md).
+The core of Next.js has been designed to enable starting as a static site (or Single-Page Application), if desired, and later upgrade to use powerful, dynamic features that require a server. For example, [Incremental Static Regeneration](/docs/basic-features/data-fetching/incremental-static-regeneration.md), [Internationalized Routing](/docs/advanced-features/i18n-routing.md), [and more](#unsupported-features).
```
Comment (styfle) [inline docs/advanced-features/static-html-export.md:16]:
```suggestion
The core of Next.js has been designed to enable starting as a static site (or Single-Page Application), if desired, and later upgrade to use powerful, dynamic features that require server-side rendering. For example, [Incremental Static Regeneration](/docs/basic-features/data-fetching/incremental-static-regeneration.md), [Internationalized Routing](/docs/advanced-features/i18n-routing.md), [and more](#unsupported-features).
```

File: docs/advanced-features/static-html-export.md
Comment (zbrogz) [inline docs/advanced-features/static-html-export.md:1]: Should the file name and page title be changed as well? Maybe "static HTML" -> "static asset"?

**Pull request merged**